### PR TITLE
Secrets deletion config

### PR DESCRIPTION
**Pull request by dmotte**
This PR introduces a configuration option to provide a way to disable the **deletion** of Kubernetes **secrets**: `enable_secrets_deletion` (default `true`).

Ratio: see https://github.com/zalando/postgres-operator/issues/1736#issuecomment-1999567249

I think many people would love this feature. This also relates to #2579, which otherwise wouldn't make much sense, because attaching already-existing volumes to a brand new cluster doesn't work due to different secrets.

Inspired by #2579 (huge thanks to @FxKu)

### CODE REVIEW

File: e2e/tests/test_e2e.py [hunk 1]
```diff
@@ -2109,7 +2110,7 @@ def test_zz_cluster_deletion(self):
             self.eventuallyEqual(lambda: k8s.count_statefulsets_with_label(cluster_label), 0, "Statefulset not deleted")
             self.eventuallyEqual(lambda: k8s.count_deployments_with_label(cluster_label), 0, "Deployments not deleted")
             self.eventuallyEqual(lambda: k8s.count_pdbs_with_label(cluster_label), 0, "Pod disruption budget not deleted")
-            self.eventuallyEqual(lambda: k8s.count_secrets_with_label(cluster_label), 0, "Secrets not deleted")
+            self.eventuallyEqual(lambda: k8s.count_secrets_with_label(cluster_label), 3, "Secrets were deleted although disabled in config")
```
Comment (FxKu) [inline e2e/tests/test_e2e.py:2113]:
```suggestion
            self.eventuallyEqual(lambda: k8s.count_secrets_with_label(cluster_label), 7, "Secrets were deleted although disabled in config")
```

There are more secrets created during e2e tests.

**Comment by FxKu** [comment]
:+1:

**Comment by idanovinda** [comment]
👍

**Comment by FxKu** [comment]
Thanks @dmotte

**Comment by Yingrjimsch** [comment]
@dmotte as far as I've understood this configuration is per operator. Is there a way to let the postgres cluster decide if it should keep the secret?

**Comment by dmotte** [comment]
@Yingrjimsch I think it can be implemented by adding a config option to the cluster manifest. To be honest I don't need it, but, if you want to give it a try, feel free to add me as a reviewer :)

**Comment by Yingrjimsch** [comment]
@dmotte I've added the functionallity [here](https://github.com/zalando/postgres-operator/pull/2750) and mentioned you in the pull request, it is fairly simple but effective for our use case. If you have any suggestions just let me know :)

**Pull request merged**